src/document.js

```diff
@@ -648,6 +656,43 @@ const documentsMain = {
 		PostMessages.sendWOPIPostMessage('loolframe', 'Action_Mention', { list })
 	},
 
+	openLinkPicker() {
+		getLinkWithPicker(null, true)
+			.then(link => {
+				PostMessages.sendWOPIPostMessage('loolframe', 'Action_InsertLink', { url: link })
+			})
+			.catch(error => {
+				console.error('Link picker promise rejected:', error)
+			})
+	},
+
+	async resolveLink(url) {
+		try {
+			const result = await axios.get(generateOcsUrl('references/resolve', 2), {
+				params: {
+					reference: url,
+				},
+			})
+			const resolvedLink = result.data.ocs.data.references[url]
+			const title = resolvedLink?.openGraphObject?.name
+			const thumbnailUrl = resolvedLink?.openGraphObject?.thumb
+			let b64Image = null
+			if (thumbnailUrl) {
+				try {
+					const imageResponse = await axios.get(thumbnailUrl)
+					if (imageResponse?.status === 200 && imageResponse?.data) {
+						b64Image = btoa(unescape(encodeURIComponent(imageResponse.data)))
+					}
+				} catch (e) {
+					console.error('Error loading the reference image', e)
+				}
+			}
+			PostMessages.sendWOPIPostMessage('loolframe', ' Action_GetLinkPreview_Resp', { url, title, image: b64Image })
+		} catch (e) {
+			console.error('Error resolving a reference', e)
```

Comment:
Can we show a toast also here so the user knows that something failed?

Reply:
:+1: